src/m365/teams/commands/conversationmember/conversationmember-add.ts

```diff
@@ -0,0 +1,282 @@
+      },
+      {
+        option: '--userId [userId]'
+      },
+      {
+        option: '--userDisplayName [userDisplayName]'
+      },
+      {
+        option: '--owner'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+    if (args.options.teamId && args.options.teamName) {
+      return 'Specify either teamId or teamName, but not both.';
+    }
+
+    if (!args.options.teamId && !args.options.teamName) {
+      return 'Specify teamId or teamName, one is required';
+    }
+
+    if (args.options.teamId && !Utils.isValidGuid(args.options.teamId as string)) {
+      return `${args.options.teamId} is not a valid GUID`;
+    }
+
+    if (args.options.channelId && args.options.channelName) {
+      return 'Specify either channelId or channelName, but not both.';
+    }
+
+    if (!args.options.channelId && !args.options.channelName) {
+      return 'Specify channelId or channelName, one is required';
+    }
+
+    if (args.options.channelId && !Utils.isValidTeamsChannelId(args.options.channelId as string)) {
+      return `${args.options.channelId} is not a valid Teams ChannelId`;
+    }
+
+    if (args.options.userId && args.options.userDisplayName) {
+      return 'Specify either userId or userDisplayName, but not both.';
+    }
+
+    if (!args.options.userId && !args.options.userDisplayName) {
+      return 'Specify userId or userDisplayName, one is required';
+    }
+
+    return true;
+  }
+
+  private getTeamId(args: CommandArgs): Promise<string> {
+    if (args.options.teamId) {
+      return Promise.resolve(args.options.teamId);
+    }
+
+    return new Promise<string>((resolve: (channelId: string) => void, reject: (error: any) => void): void => {
+      const teamRequestOptions: any = {
+        url: `${this.resource}/beta/groups?$filter=resourceProvisioningOptions/Any(x:x eq 'Team') and displayName eq '${encodeURIComponent(args.options.teamName as string)}'`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        responseType: 'json'
+      };
+
+      request
+        .get<{ value: Team[] }>(teamRequestOptions)
+        .then(response => {
+          const teamItem: Team | undefined = response.value[0];
+
+          if (!teamItem) {
+            return reject(`The specified team '${args.options.teamName}' does not exist in Microsoft Teams`);
+          }
+
+          if (response.value.length > 1) {
+            return reject(`Multiple Microsoft Teams with name '${args.options.teamName}' found. Please disambiguate:${os.EOL}${response.value.map(x => `- ${x.id}`).join(os.EOL)}`);
+          }
+
+          return resolve(teamItem.id);
+        }, err => { reject(err) });
+    })
+  }
+
+  private getChannelId(teamId: string, args: CommandArgs): Promise<string> {
+    if (args.options.channelId) {
+      const channelIdRequestOptions: any = {
```

Comment:
Do we need to validate if the particular channel exists? I don't think we're doing this in other commands and if user specifies an incorrect ID, the command will fail either way, correct?

Reply:
Right. I assumed the behavior of the [Add member](https://docs.microsoft.com/en-us/graph/api/team-post-members?view=graph-rest-1.0&tabs=http) endpoint would be similar to that of [List member](https://docs.microsoft.com/en-us/graph/api/team-list-members?view=graph-rest-1.0&tabs=http) and thought that this would be needed as mentioned at https://github.com/pnp/cli-microsoft365/issues/1880#issuecomment-720141229. But I just checked that the Add member endpoint does return an error when an incorrect channel ID is used, so I'll remove this section from the code 👍